x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/results/LegacyTextEmbeddingResults.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.results;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.inference.InferenceResults;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Writes a text embedding result in the following json format
+ * {
+ *     "text_embedding": [
+ *         {
+ *             "embedding": [
+ *                 0.1
+ *             ]
+ *         },
+ *         {
+ *             "embedding": [
+ *                 0.2
+ *             ]
+ *         }
+ *     ]
+ * }
+ * @deprecated use {@link TextEmbeddingResults} instead
+ */
+@Deprecated
+public record LegacyTextEmbeddingResults(List<Embedding> embeddings) implements InferenceResults {
+    public static final String NAME = "text_embedding_results";
```

Comment:
This is the name that was used in the openai PR [here](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/results/TextEmbeddingResults.java#L43)